src/main/java/ch/njol/skript/events/EvtStatisticChange.java

```diff
@@ -0,0 +1,134 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.registrations.EventValues;
+import ch.njol.skript.util.Getter;
+import org.bukkit.Statistic;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerStatisticIncrementEvent;
+import org.bukkit.inventory.ItemStack;
+import org.eclipse.jdt.annotation.Nullable;
+
+public class EvtStatisticChange extends SkriptEvent {
+	static {
+		Skript.registerEvent("Statistic Change", EvtStatisticChange.class, PlayerStatisticIncrementEvent.class,
+				"[player] statistic[s] (change|increase) [of %string%]")
+				.description("Called when a player statistic is incremented.",
+						"This event is not called for some high frequency statistics, e.g. movement based statistics.",
+						"You can use past/future event-number to the get the past/future new value, event-number is the difference between the old and the new value."
+				)
+				.examples(
+						"on player statistic increase:",
+						"\tsend \"Statistic increased: %event-string% from %past event-number% to %future event-number% (diff: %event-number%)\" to player",
+						"\tif event-entitytype is set:",
+						"\t\tsend \"Of entity: %event-entitytype%\" to player",
+						"\telse if event-item is set:",
+						"\t\tsend \"Of item: %event-item%\" to player",
+						"on player statistic increase of \"CHEST_OPENED\":",
+						"\tsend \"You just opened a chest!\" to player",
+						"",
+						"# 'TIME_SINCE_REST' is called too many times therefore it will not trigger the event unless specified like below",
+						"on player statistic increase of \"TIME_SINCE_REST\":",
+						"\tsend \"Your chat is being flooded\" to player"
+				)
+				.since("INSERT VERSION");
+
+		EventValues.registerEventValue(PlayerStatisticIncrementEvent.class, String.class, new Getter<String, PlayerStatisticIncrementEvent>() {
```

Comment:
I have ask pickled about this as well, I didn't do this originally because there are many statistics and registering them all as `PLAYER_PLAY_TIME` with underscores is weird. We did that for sounds as well we kept them as strings due to the same thing, so it's preferred to keep it as strings and not adding a classinfo for this, maybe later.

Reply:
I don't see an issue with it being a ClassInfo. If you're referring to .lang files? There isn't that many Statistics, it can all be written out. I'll do it if you don't want to.

Reply:
Due to the names of statistics they can lead to either parser conflicts or use conflicts see this for example
`TIME_SINCE_REST` -> `time since rest` this would be really confusing to both parser and user, this is only an example of many that's why we shouldn't make them classinfos

Reply:
Parser won't have issues when you use the .lang file. Using classinfos will only parse the string input once rather than doing it a few times. It's better to be using a ClassInfo for this. It's an enum and not as many as Sound.

Reply:
Maybe parser won't be an issue based on what you say but it will be confusing for users a statistic such as the one I mentioned above, IMO it shouldn't be added an a classinfo but if you want to you can ask the rest of the team about it, you can do it in another PR

Reply:
I don't see any confusion here.

Reply:
e.g. `time since rest` & `time since death` & see https://docs.skriptlang.org/expressions.html#ExprTimeSince
e.g. `player kills`
e.g. `FISH_CAUGHT -> fish caught` and `caught fish` type of fish event